apps/aecore/src/aec_spend_tx.erl

```diff
@@ -125,79 +125,27 @@ recipient_id(#spend_tx{recipient_id = RecipientId}) ->
 amount(#spend_tx{amount = Amount}) ->
     Amount.
 
-resolve_recipient_pubkey(Tx, Trees) ->
-    case resolve_recipient(Tx, Trees) of
-        {id, Id} ->
-            {_IdType, Pubkey} = aec_id:specialize(Id),
-            {ok, Pubkey};
-        {pubkey, Pubkey} ->
-            {ok, Pubkey};
-        {error, _Rsn} = Error ->
-            Error
-    end.
-
-resolve_recipient(#spend_tx{recipient_id = RecipientId}, Trees) ->
-    case aec_id:specialize(RecipientId) of
-        {account, RecipientPubkey} ->
-            {pubkey, RecipientPubkey};
-        {oracle, RecipientPubkey} ->
-            {pubkey, RecipientPubkey};
-        {contract, RecipientPubkey} ->
-            {pubkey, RecipientPubkey};
-        %% TODO: A registered name has pointers, a pointer has a key of type binary() and
-        %% id of type aec_id:id(). To find out what id to get from all the pointers related
-        %% to the name we need the key. <<"account_pubkey">> is hard-coded and might not be present
-        %% for the given name/namehash.
-        {name, NameHash} ->
-            Key = <<"account_pubkey">>,
-            case aens:resolve_from_hash(Key, NameHash, aec_trees:ns(Trees)) of
-                {ok, Id} -> {id, Id};
-                {error, _Rsn} = Error -> Error
-            end
-    end.
-
 -spec payload(tx()) -> binary().
 payload(#spend_tx{payload = Payload}) ->
     Payload.
 
 -spec check(tx(), aec_trees:trees(), aetx_env:env()) -> {ok, aec_trees:trees()} | {error, term()}.
-check(#spend_tx{} = SpendTx, Trees, Env) ->
-    Height = aetx_env:height(Env),
-    Checks = [fun check_sender_account/3],
-    case aeu_validation:run(Checks, [SpendTx, Trees, Height]) of
-        ok ->
-            case resolve_recipient_pubkey(SpendTx, Trees) of
-                {ok, RecipientPubkey} ->
-                    {ok, aec_trees:ensure_account(RecipientPubkey, Trees)};
-                {error, _} = E ->
-                    E
-            end;
-        {error, _Reason} = Error ->
-            Error
-    end.
+check(#spend_tx{}, Trees,_Env) ->
```

Comment:
I see you leave check check functions - where they are used?

Reply:
They are used from the `aetx.erl` still. There are a few channel transactions that are not migrated, but I left them out since they are not yet used from contracts.